tests/posix/threads_base/testcase.yaml

```diff
@@ -0,0 +1,32 @@
+common:
+  tags:
+    - posix
+    - threads_base
+  min_ram: 64
+  timeout: 240
+  # 1 tier0 platform per supported architecture
+  platform_key:
+    - arch
+    - simulation
+  filter: not CONFIG_NATIVE_LIBC
+tests:
+  portability.posix.threads_base: {}
+  portability.posix.threads_base.minimal:
+    extra_configs:
+      - CONFIG_MINIMAL_LIBC=y
+  portability.posix.threads_base.newlib:
+    filter: TOOLCHAIN_HAS_NEWLIB == 1
+    extra_configs:
+      - CONFIG_NEWLIB_LIBC=y
+  portability.posix.threads_base.picolibc:
+    tags: picolibc
+    filter: CONFIG_PICOLIBC_SUPPORTED
+    extra_configs:
+      - CONFIG_PICOLIBC=y
+  portability.posix.threads_base.dynamic_stack:
+    extra_configs:
+      - CONFIG_DYNAMIC_THREAD=y
+      - CONFIG_THREAD_STACK_INFO=y
+      - CONFIG_HEAP_MEM_POOL_SIZE=16384
+      - CONFIG_POSIX_THREAD_KEYS_MAX=2048
+      - CONFIG_TEST_EXTRA_STACK_SIZE=16384
```

Comment:
I think this case should be removed and the dynamic thread stuff should just be moved to `prj.conf`, specifically, because `pthread_setstack()` et al (for using static thread stacks) should be moved out to the `xsi_threads_ext` test, to be consistent with [Subprofiling Considerations](https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_subprofiles.html)